internal/provider/network/isolated_resource.go

```diff
@@ -0,0 +1,670 @@
+	org, _, err = r.client.GetOrgAndVDC(r.client.GetOrg(), r.client.GetDefaultVDC())
+	if err != nil {
+		resp.Diagnostics.AddError("[UPDATE] Error retrieving Org", err.Error())
+		return
+	}
+
+	// Get network
+
+	orgNetwork, err = org.GetOpenApiOrgVdcNetworkById(state.ID.ValueString())
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			// Network not found, so remove from state
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("[UPDATE] Error retrieving isolated network", err.Error())
+		return
+	}
+
+	// Set network type
+	ipPools := []staticIPPoolResourceModel{}
+	diags := plan.StaticIPPool.ElementsAs(ctx, &ipPools, true)
+	resp.Diagnostics.Append(diags...)
+	myshared := false // Cloudavenue does not support shared networks
+	networkType := &govdctypes.OpenApiOrgVdcNetwork{
+		ID:          orgNetwork.OpenApiOrgVdcNetwork.ID,
+		Name:        plan.Name.ValueString(),
+		Description: plan.Description.ValueString(),
+		OwnerRef:    orgNetwork.OpenApiOrgVdcNetwork.OwnerRef,
+		Shared:      &myshared,
+		NetworkType: govdctypes.OrgVdcNetworkTypeIsolated,
+		Subnets: govdctypes.OrgVdcNetworkSubnets{
+			Values: []govdctypes.OrgVdcNetworkSubnetValues{
+				{
+					Gateway:      plan.Gateway.ValueString(),
+					PrefixLength: int(plan.PrefixLength.ValueInt64()),
+					IPRanges: govdctypes.OrgVdcNetworkSubnetIPRanges{
+						Values: myProcessIPRanges(ipPools),
+					},
+					DNSServer1: plan.PrimaryDNS.ValueString(),
+					DNSServer2: plan.SecondaryDNS.ValueString(),
+					DNSSuffix:  plan.SuffixDNS.ValueString(),
+				},
+			},
+		},
+	}
+
+	// Check if VdcGroup or Vdc
+	var urn string
+	vdc, err := org.GetVDCById(state.OwnerID.ValueString(), true)
+	if err == nil { // It is a VDC
+		urn, _ = govcd.BuildUrnWithUuid("urn:vcloud:vdc:", vdc.Vdc.ID)
+	} else { // It is not a VDC, so it may be a VDC Group
+		urn, err := govcd.BuildUrnWithUuid("urn:vcloud:vdcGroup:", state.OwnerID.ValueString())
+		if err == nil {
+			_, err := org.GetVdcGroupById(urn)
+			if err != nil {
+				resp.Diagnostics.AddError("[UPDATE] Error retrieving VDC Group from OwnerID", err.Error())
+				return
+			}
+		} else {
+			resp.Diagnostics.AddError("[UPDATE] Error retrieving VDC or VDC Group from OwnerID", err.Error())
+			return
+		}
+	}
+
+	// Lock VDC or VDC Group to prevent concurrent access
+	vcdMutexKV := mutex.NewKV()
+	vcdMutexKV.KvLock(ctx, urn)
+	defer vcdMutexKV.KvUnlock(ctx, urn)
+
+	// Update network
+	_, err = orgNetwork.Update(networkType)
+	if err != nil {
+		resp.Diagnostics.AddError("Error updating isolated network", err.Error())
+		return
+	}
+
+	plan = &networkIsolatedResourceModel{
+		ID:           types.StringValue(orgNetwork.OpenApiOrgVdcNetwork.ID),
+		OwnerID:      state.OwnerID,
+		Name:         plan.Name,
+		Description:  plan.Description,
+		Gateway:      plan.Gateway,
+		PrefixLength: plan.PrefixLength,
+		PrimaryDNS:   plan.PrimaryDNS,
+		SecondaryDNS: plan.SecondaryDNS,
+		SuffixDNS:    plan.SuffixDNS,
+	}
```

Comment:
ID never change after Update. Use planmodifier `UseStateForUnknown` for using state if value is Unknown

For OwnerID if you use state.OwnerID in your code it is better to use planmodifier `UseStateForUnknown`

```suggestion

```